### ISSUE TITLE
principled experimentation with prompting

### ISSUE THREAD

**Issue by rbroc** [rbroc/echo#12]
- Create a few different prompt options (2 to 5) for each task
- Qualitatively inspect how models behave with each of these
- Run generation at scale with all prompt options
- Run text through text descriptives to look at overall distributional differences (how to do the comparison is an open question, see #13)

**Comment by MinaAlmasi** [comment]
Current prompts are below. We have so far experimented mostly with `falcon-instruct-7b` and `FLAN T5` (`base` and through to `XL`). `StableBeluga` also works, but we have not tried all prompts yet. 

**Currently, prompting does not seem to affect `Flan T5` at all** (whereas `falcon-instruct-7b` does seem to be affected). `T5` is mostly just topic-dependent, creating coherent and intelligible texts for topics such as "poetry" but otherwise returning repetitive text (e.g., only repeating one word or repeating the prompt). We should consider whether we want to include `T5` at all (we can make this decision after trying `T5 XXL`, but so far we have not seen major improvements when increasing the size).

Running at scale has not been done yet as we wanted to have a more solid pipeline before burning GPU compute !! 
```
 prompts = {
            # daily mail (summarization)
            "dailymail_cnn_1": "summarize the main points of this article: ",
            "dailymail_cnn_2": "create a summary of the news article: ",
            "dailymail_cnn_3": "write a short summarized text of the news article: ",
            "dailymail_cnn_4": "summarize this: ",

            # stories (text generate)
            "stories_1": "continue the story: ",
            "stories_2": "write a small text based on this story: ",
            "stories_3": "complete the text: ",
            "stories_4": "complete the story: ",

            # mrpc (paraphrase)
            "mrpc_1": "paraphrase this text: ",
            "mrpc_2": "summarize this text: ",
            "mrpc_3": "summarize this: ",
            "mrpc_4": "create a summary of this: ",

            # dailydialog
            "dailydialog_1": "respond to the final sentence: ",
            "dailydialog_2": "continue this dialog: ",
            "dailydialog_3": "finish this dialog: ",
            "dailydialog_4": "respond to the final sentence: ",
        }

```

**Comment by rbroc** [comment]
re: our fire evacuation conversation, I've skimmed through the FLAN-T5 paper, and they do train some versions of the models doing one-shot. I've also found this blog post for summarization w/ Flan-T5: https://blog.gopenai.com/text-summarization-using-flan-t5-5ded2e4ce182.

**Comment by rbroc** [comment]
(this is more of a note to self: there's a separate paper doing experimentation with prompting specifically for FLAN-T5, and templates used for experimentation can be found here: https://github.com/google-research/FLAN/blob/main/flan/v2/flan_templates_branched.py)

**Comment by MinaAlmasi** [comment]
> re: our fire evacuation conversation, I've skimmed through the FLAN-T5 paper, and they do train some versions of the models doing one-shot. I've also found this blog post for summarization w/ Flan-T5: https://blog.gopenai.com/text-summarization-using-flan-t5-5ded2e4ce182.

Interesting! We'll take a look !

**Comment by rbroc** [comment]
closed by #48
